src/lib/verifier/generic.ml

```diff
@@ -0,0 +1,9 @@
+open Core_kernel
+open Async_kernel
+
+type 'ledger_proof t =
```

Comment:
why not a record? Seems counter to style in the rest of the codebase and you don't seem to be taking advantage of any object-specific capability